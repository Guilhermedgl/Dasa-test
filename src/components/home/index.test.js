import React from 'react';
import { create } from 'react-test-renderer';
import Home from './index';

describe('<Home />', () => {
  const pokemonsProps = [{
    name: 'bulbasaur'
  },
  {
    name: 'ivysaur'
  },
  {
    name: 'venusaur'
  },
  {
    name: 'charmander'
  },
  {
    name: 'charmeleon'
  }];

  it('should match with snapshot', () => {
    const component = create(<Home pokemons={pokemonsProps} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should have a empty string in initial state', () => {
    const component = create(<Home pokemons={pokemonsProps} />);
    const instance = component.getInstance();
    expect(instance.state.search).toHaveLength(0);
  });

  it('should return full array with empty search', () => {
    const component = create(<Home pokemons={pokemonsProps} />);
    const instance = component.getInstance();
    expect(instance.renderList()).toEqual(pokemonsProps);
  });

  it('should filter the array with e.target.value', () => {
    const event = {
      preventDefault: () => {
        return;
      },
      target: {
        value: 'blb'
      }
    };

    const component = create(<Home pokemons={pokemonsProps} />);
    const instance = component.getInstance();
    instance.handleChange(event);
    expect(instance.renderList()).toEqual([{ name: 'bulbasaur' }])
  });
});