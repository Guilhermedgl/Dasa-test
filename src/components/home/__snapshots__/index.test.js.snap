// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should match with snapshot 1`] = `
<section>
  <nav
    className="navbar"
  >
    <h1>
      Pokemons
    </h1>
  </nav>
  <div
    className="container"
  >
    <div
      className="container-input"
    >
      <input
        className="input"
        onChange={[Function]}
        placeholder="Pesquise por um Pokemon"
        type="text"
      />
    </div>
    <div>
      <h3
        className="notes"
      >
        Notas
      </h3>
      <div
        className="card"
      >
        <p
          className="name"
        >
          bulbasaur
        </p>
      </div>
      <div
        className="card"
      >
        <p
          className="name"
        >
          ivysaur
        </p>
      </div>
      <div
        className="card"
      >
        <p
          className="name"
        >
          venusaur
        </p>
      </div>
      <div
        className="card"
      >
        <p
          className="name"
        >
          charmander
        </p>
      </div>
      <div
        className="card"
      >
        <p
          className="name"
        >
          charmeleon
        </p>
      </div>
    </div>
  </div>
</section>
`;
